project-base/storefront/hooks/cart/useReloadCart.ts

```diff
@@ -16,12 +18,22 @@ export const useReloadCart = (): void => {
     const isUserLoggedIn = useIsUserLoggedIn();
     const cartUuid = usePersistStore((store) => store.cartUuid);
     const client = useClient();
+    const slug = getUrlWithoutGetParameters(router.asPath);
 
     useEffect(() => {
-        if (cartUuid || isUserLoggedIn) {
-            client.query(CartQueryDocumentApi, { cartUuid }, { requestPolicy: 'network-only' }).toPromise();
-        }
-    }, [router.asPath]);
+        const func = async () => {
+            const cookies = getCookies();
+            const isWithUserTokens = !!(cookies.accessToken && cookies.refreshToken);
+
+            if ((isUserLoggedIn && !isWithUserTokens) || (!isUserLoggedIn && isWithUserTokens)) {
+                router.reload();
+            } else if (cartUuid || isUserLoggedIn) {
+                client.query(CartQueryDocumentApi, { cartUuid }, { requestPolicy: 'network-only' }).toPromise();
+            }
+        };
+
+        func();
+    }, [slug]);
```

Comment:
Seems like it doesnt have to be async and can be thus simplified

Reply:
Yup, I forgot this after some debugging. Refactored.